crates/oxc_ast/src/ast/ts.rs

```diff
@@ -119,21 +119,38 @@ pub struct TSEnumBody<'a> {
 #[ast(visit)]
 #[derive(Debug)]
 #[generate_derive(CloneIn, Dummy, TakeIn, GetSpan, GetSpanMut, ContentEq, ESTree)]
+#[estree(
+    add_fields(computed = TSEnumMemberComputed),
+    field_order(span, id, computed, initializer),
+)]
 pub struct TSEnumMember<'a> {
     pub span: Span,
     pub id: TSEnumMemberName<'a>,
-    pub computed: bool,
     pub initializer: Option<Expression<'a>>,
 }
 
 /// TS Enum Member Name
+///
+/// ## Example
+/// ```ts
+/// enum ValidEnum {
+///   identifier,
+///   'string',
+///   ['computed-string'],
+///   [`computed-template`],
+///   // These are invalid syntax
+///   // `template`,
+///   // [computedIdentifier],
```

Comment:
Turns out I was wrong about this. `[computedIdentifier]` causes a *type check* error, but not a *parser* error, so I think we do need to support it.

Please see bottom of my comment on #10332.

Reply:
If you put that as 3rd variant (after `String`), the `raw_deser` implementation I've added will remain valid. If you want to put the variants in a different order, you'd need to change the `raw_deser` implementation.

Reply:
> [computedIdentifier] causes a type check error, but not a parser error

Ah, I finally understand!

It turns out that we should fix the current parser implementation, doesn't it?

https://github.com/oxc-project/oxc/blob/63d4aa6ac5cc552712071349059194e6e07f960b/crates/oxc_parser/src/ts/statement.rs#L82

> I'm not sure what static_name should return for ComputedIdentifier. None?

Indeed... 🤔 First, let me check all usages of `static_name'.

Reply:
> It turns out that we should fix the current parser implementation, doesn't it?

I think so, yes.

Reply:
After some research, I found these are also the same category. 😓 

```ts
 const x = 'a';
 enum Foo { [x] }

// Expression
 enum Bar { ['a' + 'b'] }
```

Reply:
Good find! Yes, TS-ESLint also parses that without complaint: [AST explorer](https://ast-explorer.dev/#eNolyjEOwjAMBdCrWH/JAFygI4dgwQwm8hCUulESkFCUu9dV16c3kLHgIz9psabScUVx6P+iJ9y05WSHR3e170p3qTToGSTQhcI7vGiyedg8DDYihsZVHlpb2oyxOGTp2jqDbWLul5olMg==)

OK, this is getting too complicated. Let's ignore the whole of this comment chain, and leave it as you have it in this PR. If we find that this fails some of TS-ESLint conformance tests, then we can revisit the decision.

Sorry I have wasted time with this red herring!

Reply:
> Sorry I have wasted time with this red herring! But thank you for delving into it deeply.

Not at all! Thank you for bringing it up. I appreciate it~😃

Reply:
NOTE: For the future...

- `TSEnumMember`.`id`: `PropertyNameComputed | PropertyNameNonComputed`
  - https://github.com/typescript-eslint/typescript-eslint/blob/7f23300d0d23e5abbb7b86f8c40bafd7730901e1/packages/ast-spec/src/element/TSEnumMember/spec.ts#L12-L14
- Computed = `Expression`, NonComputed = `Identifier` + `StringLiteral` + `NumberLiteral`
  - https://github.com/typescript-eslint/typescript-eslint/blob/7f23300d0d23e5abbb7b86f8c40bafd7730901e1/packages/ast-spec/src/unions/PropertyName.ts#L11-L15